### ISSUE TITLE
Support using non-default Swift toolchains

### ISSUE THREAD

**Issue by grangej** [realm/jazzy#743]
I am seeing this error but when i run the same xcodebuild I get no errors from Xcode. I get 0% documentation. 

`A compile error prevented NotificationURLLinkDelegate.viewControllerForLinkURL(notification:) from receiving a unique USR. Documentation may be incomplete. Please check for compile errors by running `xcodebuild`

**Comment by jpsim** [comment]
@grangej what are the steps to reproduce this?

**Comment by grangej** [comment]
@jpsim I am running against a pure swift framework. I have tried both with and without specifying the Xcode build options. 

Currently I have the following build : 

jazzy -x -workspace,IdentityOperations.xcworkspace,-scheme,IdentityOperations

**Comment by jpsim** [comment]
can you share the affected project? (IdentityOperations)

**Comment by grangej** [comment]
Unfortunately not , I will try to reproduce in something simple.

**Comment by grangej** [comment]
Ok, I have reproduced error with a simple framework. 

https://github.com/grangej/Logger


Here is the output:

`Running xcodebuild
Parsing Logger.swift (1/5)
Parsing CrashlyticsRecorder.swift (2/5)
Parsing ErrorProtocal.swift (3/5)
Parsing SumoLogger.swift (4/5)
Parsing SuplexLogger.swift (5/5)
A compile error prevented Logger.log(logMessage:logPrefix:) from receiving a unique USR. Documentation may be incomplete. Please check for compile errors by running `xcodebuild `.
A compile error prevented ErrorType.logError() from receiving a unique USR. Documentation may be incomplete. Please check for compile errors by running `xcodebuild `.
A compile error prevented ErrorType.logWarning() from receiving a unique USR. Documentation may be incomplete. Please check for compile errors by running `xcodebuild `.
A compile error prevented ErrorType.logInfo() from receiving a unique USR. Documentation may be incomplete. Please check for compile errors by running `xcodebuild `.
A compile error prevented ErrorType.log(logLevel:) from receiving a unique USR. Documentation may be incomplete. Please check for compile errors by running `xcodebuild `.
building site
jam out ♪♫ to your fresh new docs in `docs`
C02R30HWG8WM-L:logger john.grange$ xcodebuild 
=== BUILD TARGET Logger OF PROJECT Logger WITH THE DEFAULT CONFIGURATION (Release) ===

Check dependencies

CodeSign build/Release-iphoneos/Logger.framework
    cd /Users/john.grange/Projects/Lifelock/logger
    export CODESIGN_ALLOCATE=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/codesign_allocate
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    
Signing Identity:     "iPhone Developer: Dev XcodeBuildServer (KEEKA7JA8S)"

    /usr/bin/codesign --force --sign E83B5419176425CA204BBF2DC916248B6B811191 --timestamp=none /Users/john.grange/Projects/Lifelock/logger/build/Release-iphoneos/Logger.framework

** BUILD SUCCEEDED **
`

**Comment by jpsim** [comment]
Looks like you're trying to use a Swift 3 toolchain to build a Swift 2 codebase...

Try specifying the correct Swift toolchain:

```
$ TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 jazzy
Running xcodebuild
Parsing Logger.swift (1/5)
Parsing CrashlyticsRecorder.swift (2/5)
Parsing ErrorProtocal.swift (3/5)
Parsing SumoLogger.swift (4/5)
Parsing SuplexLogger.swift (5/5)
30% documentation coverage with 39 undocumented symbols
building site
building search index
jam out ♪♫ to your fresh new docs in `docs`
```

**Comment by grangej** [comment]
@jpsim ! Thank you ! Can I specify this in the .jazzy.yaml ?

**Comment by jpsim** [comment]
Once https://github.com/segiddins/xcinvoke/pull/5 is merged, you'll be able to do this:

```yaml
swift_version: 2.3
```

**Comment by x-du** [comment]
@grangej did you make it work?

**Comment by grangej** [comment]
No, this PR still appears to be open.

**Comment by rajeevsharma2811** [comment]
What is the status of this issue?

**Comment by johnfairh** [comment]
This issue is tracking the rare scenario where the required swift toolchain has to be specified manually (in the example above, Swift 2.3 instead of default Swift 3).  If your project does not have that problem (with the workaround/fix above) then please raise a new issue.

**Comment by johnfairh** [comment]
Closing in favor of older dup #610.
